postmaster/apiv1/admins.py

```diff
@@ -147,3 +149,115 @@ def unlock_admin(admin_id):
+                    'The following error occurred in twofactor_disable: {0}'.format(str(e)))
+                raise GenericError('The administrator could not be updated')
+            return dict(enabled=admin.otp_active)
+        elif status.lower() == "true":
+            raise GenericError("Cannot enable 2 factor from this route - see docs")
+    raise GenericError("An Invalid parameter was supplied")
+
+
+@apiv1.route('/admins/<int:admin_id>/2factor/qrcode', methods=['GET'])
+@login_required
+def qrcode(admin_id):
+    """ Presents the user with a QR code to scan to setup 2 factor authentication
+    """
+    # render qrcode for FreeTOTP
+    admin = Admins.query.get_or_404(admin_id)
+    if admin.id != current_user.id:
+        raise GenericError('You may not view other admin\'s QR codes')
+    if admin.otp_active:
+        return ('', 204)
+    admin.generate_otp_secret()
+    try:
+        db.session.add(admin)
+        db.session.commit()
+    except ValidationError as e:
+        raise e
+    except Exception as e:
+        db.session.rollback()
+        json_logger(
+            'error', current_user.username,
+            'The following error occurred in qrcode: {0}'.format(str(e)))
+        raise GenericError('The administrator could not be updated')
+    url = pyqrcode.create(admin.get_totp_uri())
+    stream = StringIO()
+    url.svg(stream, scale=5)
+    return stream.getvalue().encode('utf-8'), 200, {
+        'Content-Type': 'image/svg+xml',
+        'Cache-Control': 'no-cache, no-store, must-revalidate',
+        'Pragma': 'no-cache',
+        'Expires': '0'}
+
+
+@apiv1.route('/admins/<int:admin_id>/2factor/verify', methods=['POST'])
+@login_required
+@json_wrap
+def verify_qrcode(admin_id):
+    """ Verifies if the 2 factor token provided is correct
+
+        This will enable 2 factor for a user.
+    """
+    admin = Admins.query.get_or_404(admin_id)
+    if request.get_json(force=True).get('code'):
+        if not admin.otp_secret:
+            raise GenericError("The 2 Factor Secret has not been generated yet")
+        if admin.verify_totp(request.get_json(force=True).get('code')):
+            if not admin.otp_active:
+                audit_message = 'The administrator "{0}" enabled 2 factor'.format(
+                    admin.username)
+                admin.otp_active = True
+                try:
+                    db.session.add(admin)
+                    db.session.commit()
+                    json_logger('audit', current_user.username, audit_message)
+                except ValidationError as e:
+                    raise e
+                except Exception as e:
+                    db.session.rollback()
+                    json_logger(
+                        'error', current_user.username,
+                        'The following error occurred in verify_qrcode: {0}'.format(str(e)))
+                    raise GenericError('The administrator could not be updated')
+            return dict(status="Success")
```

Comment:
I don't think we should be returning this here. We don't anywhere else in the app. A simple HTTP 200 should do here with no JSON.